test/rateLimiters/fixedWindow.test.ts

```diff
@@ -0,0 +1,128 @@
+                expect(tokenCountPartial.currentTokens).toBe(initial + partialWithdraw);
+            });
+
+            test('window is partially full and request has no leftover tokens', async () => {
+                const initial = 6;
+                const partialWithdraw = 5;
+                await setTokenCountInClient(client, user2, initial, timestamp);
+                expect(
+                    (await limiter.processRequest(user2, timestamp, partialWithdraw)).success
+                ).toBe(false);
+                expect(
+                    (await limiter.processRequest(user2, timestamp, partialWithdraw)).tokens
+                ).toBe(0);
+            });
+        });
+        describe('after a BLOCKED request...', () => {
+            afterEach(() => {
+                client.flushall();
+            });
+            test('initial request is greater than capacity', async () => {
+                // expect remaining tokens to be 10, b/c the 11 token request should be blocked
+                expect((await limiter.processRequest(user1, timestamp, 11)).success).toBe(false);
+                // expect current tokens in the window to still be 0
+                expect((await getWindowFromClient(client, user1)).currentTokens).toBe(0);
+            });
+            test('window is partially full but not enough time elapsed to reach new window', async () => {
+                const requestedTokens = 9;
+
+                await setTokenCountInClient(client, user2, requestedTokens, timestamp);
+                // expect remaining tokens to be 1, b/c the 2-token-request should be blocked
+                const result = await limiter.processRequest(user2, timestamp + WINDOW_SIZE - 1, 2);
+
+                expect(result.success).toBe(false);
+                expect(result.tokens).toBe(0);
+
+                // expect current tokens in the window to still be 9
+                expect((await getWindowFromClient(client, user2)).currentTokens).toBe(9);
+            });
+        });
+        describe('updateTimeWindow function works as expect', () => {
+            afterEach(() => {
+                client.flushall();
+            });
+            test('New window is initialized after reaching the window size', async () => {
+                const fullRequest = 10;
+                await setTokenCountInClient(client, user3, fullRequest, timestamp);
+                const noAccess = await limiter.processRequest(
+                    user3,
+                    timestamp + WINDOW_SIZE - 1,
+                    1
+                );
+
+                // expect cannot pass any request
+                expect(noAccess.tokens).toBe(0);
+                expect(noAccess.success).toBe(false);
+
+                const newRequest = 3;
+                await setTokenCountInClient(client, user3, newRequest, timestamp + WINDOW_SIZE + 1);
+                const newAccess = await limiter.processRequest(user3, timestamp, 1);
```

Comment:
To properly test this, don't set a new token in the cache with line 120, just process another request on line 121 but with `timestamp + WINDOW_SIZE + 1`

Reply:
`timestamp + WINDOW_SIZE` would also be appropriate. New windows begin when window size is reached

Reply:
`timestamp + WINDOW_SIZE` would also work. New windows should start when the timestamp reaches window size, so the +1 is unnecessary

Reply:
Will change, triple check is a must for this.